STYLE.md

```diff
@@ -0,0 +1,740 @@
+
+When you request multiple test runs using the `-count` flag, the intention is to run the tests multiple times. There's no point running them just once and showing the same result n-1 times. So `-count` triggers omitting the cached results.
+
+Setting `-count=1` will cause the tests to run once, omitting previously cached outputs. (The default value of `count` is 1, but you need to explicitly set the value to 1 to change the caching behavior.)
+
+### `t.Fatalf` or `t.Errorf`?
+
+If a test should not continue at a certain point, use `t.Fatalf` instead of `t.Errorf`. Do not use `t.Errorf` without thinking about it everywhere. There is a difference between them.
+
+### Ports
+
+When testing a port, use a high port that's not likely to be used. A port above 60000 would be better than 4000 or 8080.
+
+### Comments in tests
+
+All complex tests should have a verbose comment describing what they are testing.
+
+### Philosophies
+
+#### Avoid defaults
+
+Try not to have defaults for tests, as they often get in the way of additional tests.
+
+#### Check behaviour, not logs
+
+Do not bother checking the logs in the tests, because stable log formatting isn't part of the contract. Check the expected behaviour or output.
+
+#### Time-dependent tests
+
+When a test needs to wait (e.g., in a for loop or a sleep), make sure the time is long enough to ensure that the test passes even when the CPU is loaded. Also make sure the time is not excessively long, which would slow the tests drastically. Use a reasonable value, refer to existing tests, follow existing convention, and maybe run the test multiple times to get a reasonable value - do not simply set a value and leave it at that.
+
+#### Test coverage
+
+Do the tests cover all scenarios?
+
+Do all newly added functions have tests?
+
+## Comments
+
+### Be precise
+
+Think about choice of words, especially verbs. Are errors "returned" or "thrown"?
+
+### Helper/utility functions
+
+Add comments for complex helper/utility functions to describe their use.
+
+### Don't write obvious comments
+
+Either delete an obvious comment or make it more useful.
+
+For example:
+
+- Avoid: "MkdirOptions is a struct of options used for Mkdir()."
+- Prefer: Either remove the comment or: "MkdirOptions holds the options for a call to Mkdir."
+
+For another example:
+
+```go
+	// if path already exists
+	if s, err := os.Stat(path); err == nil {
+		// ...
+	}
+```
+
+The comment probably isn't necessary since the following line is straightforward to understand.
+
+### Use `TODO`
+
+Add a "TODO" in the comment when handling temporary workarounds so it's easier to grep for later.
+
+### Housekeeping rule
+
+Be careful when the code you change has some comments/notes or even links to issues. Read them carefully. If your code change solves the issue, remove the link - keep the comment clean and true.
+
+### Spell-check and grammar-check comments
+
+Do this.
+
+### Review newly added/updated comments before committing
```

Comment:
I think this goes without saying too, at least not here; all committed code and comments should be reviewed.